src/maketype.jl

```diff
@@ -424,3 +428,21 @@ function addjumps!(rn::DiffEqBase.AbstractReactionNetwork;
 
     nothing
 end
+
+function addequi!(rn::DiffEqBase.AbstractReactionNetwork)
+    @unpack params, reactions, syms_to_ints, params_to_ints, scale_noise = rn
+
+    # first construct an ODE reaction network (required for getting f_func).
+    if rn.f == nothing
+        addodes!(rn)
```

Comment:
Could this be what is causing your eval issues? Maybe require `addodes!` to be called before calling `addequi!` by the user? Then you could just throw an error here saying to call `addodes!` first if `rn.f` is not defined.

Reply:
This part is fine, I tried removing this part and calling before and the issue persisted.